Changelog.md

```diff
@@ -1,10 +1,44 @@
 # Unreleased
 
+# 0.15.0 – 2024-01-14
+
+## Breaking changes
+
+- [replace software_interrupt! macro with generic function](https://github.com/rust-osdev/x86_64/pull/259)
+- [Use SegmentSelector in InterruptStackFrame](https://github.com/rust-osdev/x86_64/pull/263)
+- [add `InvalidStarSegmentSelectors` error](https://github.com/rust-osdev/x86_64/pull/317)
+- [add `PcidTooBig` error](https://github.com/rust-osdev/x86_64/pull/316)
+- [implement `Index<u8>` for IDT instead of `Index<usize>`](https://github.com/rust-osdev/x86_64/pull/319)
+- [change `cpu_flags`'s type to `RFlags`](https://github.com/rust-osdev/x86_64/pull/324)
+- [fix `load_tss` and `GlobalDescriptorTable`](https://github.com/rust-osdev/x86_64/pull/323)
+- [add an immutable getter for the level 4 page table](https://github.com/rust-osdev/x86_64/pull/327)
+- [make `Cr2::read` return a result](https://github.com/rust-osdev/x86_64/pull/335)
+- [remove `external_asm` and `inline_asm` features](https://github.com/rust-osdev/x86_64/pull/345)
+- [Allow the GDT to be of any length](https://github.com/rust-osdev/x86_64/pull/360)
+- [Remove software_interrupt! macro](https://github.com/rust-osdev/x86_64/pull/363)
+- [Remove usize trait impls](https://github.com/rust-osdev/x86_64/pull/364)
+- [Remove deprecated functions/flags](https://github.com/rust-osdev/x86_64/pull/368)
+- [VirtAddr improvements](https://github.com/rust-osdev/x86_64/pull/370)
+- [Add structures::gdt::Entry type](https://github.com/rust-osdev/x86_64/pull/380)
+- [Allow GDT to be loaded with shared reference](https://github.com/rust-osdev/x86_64/pull/381)
+- [seal off the `PageSize` trait](https://github.com/rust-osdev/x86_64/pull/404)
+
 ## New Features
 
 - [Add `HandlerFuncType` trait](https://github.com/rust-osdev/x86_64/pull/439)
 
-# 0.14.11 – 2022-09-15
+## Fixes
+
+- [fix typo in docs](https://github.com/rust-osdev/x86_64/pull/265)
+- [activate `feature(asm_const)`](https://github.com/rust-osdev/x86_64/pull/320)
+- [gdt: Check that MAX is in range](https://github.com/rust-osdev/x86_64/pull/365)
+- [fix `Page::from_page_table_indices`](https://github.com/rust-osdev/x86_64/pull/398)
+
+## Other improvements
+
+- [idt: Fixup Options structure and cleanup set_handler_fn](https://github.com/rust-osdev/x86_64/pull/226)
```

Comment:
I'm not entirely sure why this PR was merged into the next branch. AFAICT it doesn't introduce any breaking changes and doesn't depend on any changes in the next branch.

Reply:
Looks like the behavior of `set_handler_function`/`set_handler_addr` was changed in a subtle way. Before #226, only the `present` was added to the entry options. Since #226, the whole `options` field is reinitialized, which means that the additional options are reset (if they were set before).

We should probably add the doc comment introduced in #226 to both `set_handler_function` and `set_handler_addr`:
https://github.com/rust-osdev/x86_64/pull/261/files#diff-399920f192f3fe5242eba18a5bdecd151fe662e3432e4836a34d11a7b8076b54R630-R638

Reply:
Also, this part of the `set_handler_fn` comment is no longer true:

> /// This method is only usable with the `abi_x86_interrupt` feature enabled. Without it, the
> /// unsafe [`Entry::set_handler_addr`] method has to be used instead.

The function is no longer gated on the `abi_x86_interrupt` feature, only the impls for `HandlerFunc*` are. So users can still use this function if they provide their own `HandlerFuncType` implementation.

Reply:
> if they provide their own `HandlerFuncType` implementation

Speaking of which, we should probably make the `HandlerFuncType` an `unsafe trait`, given that we trust the `to_virt_addr` output.

Reply:
Ah, I forgot that we only use the trait with specific types defined in `InterruptDescriptorTable`. So there is no reason for users to implement this trait. Still, an unsafe trait would probably better represent our intentions?

Reply:
> Also, this part of the `set_handler_fn` comment is no longer true:
> 
> > /// This method is only usable with the `abi_x86_interrupt` feature enabled. Without it, the
> > /// unsafe [`Entry::set_handler_addr`] method has to be used instead.
> 
> The function is no longer gated on the `abi_x86_interrupt` feature, only the impls for `HandlerFunc*` are. So users can still use this function if they provide their own `HandlerFuncType` implementation.

It's still only usable if `abi_x86_interrupt` is active because the `HandlerFuncType` is gated behind `abi_x86_interrupt`:
https://github.com/rust-osdev/x86_64/blob/47a11c92dbabf372cbab0da59f599d8305e0a097/src/structures/idt.rs#L839-L855